packages/magento-customer/hooks/useCustomerValidateToken.ts

```diff
@@ -0,0 +1,23 @@
+import { useApolloClient } from '@graphcommerce/graphql'
+import { ErrorCategory } from '@graphcommerce/magento-graphql'
+import { useRouter } from 'next/router'
+import { UseCustomerValidateTokenDocument } from './UseCustomerValidateToken.gql'
+import { useCustomerQuery } from './useCustomerQuery'
+
+/** Validates the current customer token. This hook is supposed to be called only once. */
+export function useCustomerValidateToken() {
+  const client = useApolloClient()
+  const router = useRouter()
+
+  useCustomerQuery(UseCustomerValidateTokenDocument, {
+    initialFetchPolicy: 'network-only',
+    onError: async ({ graphQLErrors }) => {
+      const category: ErrorCategory = 'graphql-authorization'
+      // If there is no authorization error, do nothing.
+      if (!graphQLErrors.some((e) => e.extensions?.category === category)) return
+
+      await client.clearStore()
+      router.reload()
```

Comment:
I'm not really happy this is required, but cleanStore will refetch all queries which isn't what we want because most of those queries will then fail.

Ideas welcome